src/validate.py

```diff
@@ -0,0 +1,35 @@
+from schema import Schema, And, Or, Optional
+
+# used for python2 and python3 string types
+from six import text_type
+
+def is_stringy(v):
+    return type(v) is text_type
+
+ConfigSchema = Schema({
+    "run_type": And(is_stringy,
+        lambda rt: rt in ["single", "multi", "distributed"]),
```

Comment:
This should be re-flowed so that `run_type` doesn't look like something that has `meta` as one of its keys.